datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -743,30 +749,10 @@ fn temporal_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataTyp
     use arrow::datatypes::IntervalUnit::*;
     use arrow::datatypes::TimeUnit::*;
 
-    if lhs_type == rhs_type {
-        return Some(lhs_type.clone());
-    }
     match (lhs_type, rhs_type) {
         // interval +/-
         (Interval(_), Interval(_)) => Some(Interval(MonthDayNano)),
         (Date64, Date32) | (Date32, Date64) => Some(Date64),
-        (Utf8, Date32) | (Date32, Utf8) => Some(Date32),
```

Comment:
This was split out into a separate string_temporal_coercion as it cannot be used for arithmetic